### PR TITLE
Qualify more functions as pragma(printf)

### PR DESCRIPTION
**Pull request by nordlow**
and adjust some discovered formatting errors to elide warnings.

### CODE REVIEW

**Comment by nordlow** [comment]
Can somebody help me decode the output:

�[1msrc/dmd/dtoh.d(1119): �[1;31mDeprecation: �[margument �[0;36m�[m�[1msz�[0;36m�[m for format specification �[0;36m�[m�[1m"%lu"�[0;36m�[m must be �[0;36m�[m�[1mint�[0;36m�[m, not �[0;36m�[m�[1mconst�[0;36m(�[m�[1mulong�[0;36m)�[m
�[1msrc/dmd/mtype.d(4945): �[1;31mDeprecation: �[margument �[0;36m�[m�[1msz�[0;36m�[m for format specification �[0;36m�[m�[1m"%lu"�[0;36m�[m must be �[0;36m�[m�[1mint�[0;36m�[m, not �[0;36m�[m�[1mulong�[0;36m�[m

? :)

File: src/dmd/dtoh.d [hunk 1]
```diff
@@ -1116,7 +1116,7 @@ public:
         checkbuf.level++;
         const sn = sd.ident.toChars();
         const sz = sd.size(Loc.initial);
-        checkbuf.printf("assert(sizeof(%s) == %d);", sn, sz);
+        checkbuf.printf("assert(sizeof(%s) == %lu);", sn, sz);
```
Comment (MoonlightSentinel) [inline src/dmd/dtoh.d:1119]: `sz` is a `uint64_t`, so this format should use `llu` instead of `lu`.

**Comment by MoonlightSentinel** [comment]
At least Azure prints the correct error message:
```
src\dmd\dtoh.d(1123): Deprecation: argument `sz` for format specification `"%lu"` must be `int`, not `const(ulong)`

src\dmd\mtype.d(4945): Deprecation: argument `sz` for format specification `"%lu"` must be `int`, not `ulong`

src\dmd\mtype.d(4946): Deprecation: argument `nargs - u` for format specification `"%lu"` must be `int`, not `ulong
```

File: src/dmd/dtoh.d (hunk 1 above)
Comment (nordlow) [reply src/dmd/dtoh.d]: Done.

**Comment by nordlow** [comment]
> At least Azure prints the correct error message:

Thanks. I've fixed these.

**Comment by nordlow** [comment]
What about 32-bit builds when formatting `size_t` such as in

    buf.printf("expected %llu variadic argument(s)", sz);

in mtype.d? For 64bit `%llu` is the format? Do we have to use `version`?

https://github.com/dlang/dmd/pull/11709/checks?check_run_id=1084723959

**Comment by Geod24** [comment]
For `size_t`, use `zu`

**Comment by nordlow** [comment]
> For `size_t`, use `%zu`

There are many places where `ulong(...)` is used in place of `%zu`. Shall I update those aswell?

**Review by Geod24 — APPROVED** [review]

**Comment by Geod24** [comment]
> Shall I update those as well?

That'd be nice.

**Comment by nordlow** [comment]
Is there any compiler in the CI machinery that uses version 2.092 or later?

**Comment by MoonlightSentinel** [comment]
Some Azure and Cirrus jobs use `LATEST` (currently 2.093.1)

**Pull request merged**